indico/modules/events/registration/notifications.py

```diff
@@ -32,7 +35,15 @@ def notify_invitation(invitation, email_body, from_address):
 
 
 def _notify_registration(registration, template, to_managers=False):
-    template = get_template_module('events/registration/emails/{}'.format(template), registration=registration)
+    custom_message = ''
+    if registration.state == RegistrationState.pending:
+        custom_message = registration.registration_form.message_pending
+    elif registration.state == RegistrationState.unpaid:
+        custom_message = registration.registration_form.message_unpaid
+    elif registration.state == RegistrationState.complete:
+        custom_message = registration.registration_form.message_complete
+    template = get_template_module('events/registration/emails/{}'.format(template), registration=registration,
+                                   custom_message=Markup(urlize(escape(custom_message))))
```

Comment:
I agree that it's simpler to implement, but then I would have to add it in every template, whereas this fixes it for all.